22-ajax-lab/src/01-swapi.jsx

```diff
@@ -10,19 +10,117 @@
  * allowing a user to expand from a Person to a Film she played in,
  * to another Person from that film and so on.
  */
-
 import React from 'react';
+import $ from 'jquery';
+
+const PersonDao = {};
+
+PersonDao.search = function (id) {
+    return $.getJSON(`http://swapi.co/api/people/${id}`);
+}
+
+const FilmDao = {};
+
+FilmDao.search = function (id) {
+    return $.getJSON(`http://swapi.co/api/films/${id}`);
+}
+
+export default class Person extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {person: {}, visible:false}
+        if (this.props.init)
+            this.getPersonData(this.props.id);
+    }
+
+    getPersonData()  {
+        const xhr = PersonDao.search(this.props.id);
+        xhr.then((res) => {
```

Comment:
better to change to fetch api or axios. Anyways you need to deal with request error using promise catch hook